features/layer_lock.h

```diff
@@ -82,7 +82,36 @@ void layer_lock_invert(uint8_t layer);
 //   }
 void layer_lock_set_user(layer_state_t locked_layers);
 
+#if LAYER_LOCK_IDLE_TIMEOUT > 0 // Layer Lock Timer
+
+// If you want to automatically disable layer lock if no keys are pressed after a
+// configurable timeout, define LAYER_LOCK_IDLE_TIMEOUT in your config.h file.
+// Also add the layer_lock_timer_task function under your matrix_scan_user function,
+// most likely in keymap.c
+
+// Timeout values between 100 and 32767 (in milliseconds) will use a 16 bit timer, while
+// values above 32767 will use a 32 bit timer, which uses slightly more space on
+// your MCU.
+
+    void layer_lock_timer_task(void);
```

Comment:
To follow QMK's naming conventions, please rename `layer_lock_timer_task()` to `layer_lock_task()`. Generally, a feature has an event handler called `process_<featurename>()` and, if needed, a matrix scan task called `<featurename>_task()`.

Also, define `layer_lock_task()` even when timeout is disabled as an empty no-op function. This is a good minor convenience: if the user was using timeout, but then disables it, their keymap will continue to build without having to remove the `layer_lock_task()` call in their `matrix_scan_user()` function. 

```c
// In layer_lock.h
#if LAYER_LOCK_IDLE_TIMEOUT > 0
void layer_lock_task(void);
#else
static inline void layer_lock_task(void) {}
#endif

// In layer_lock.c
#if LAYER_LOCK_IDLE_TIMEOUT > 0
void layer_lock_task(void) {
  if (locked_layers && // Check timer, etc.
}
#endif
```

I did something similar in Caps Word ([code link](https://github.com/getreuer/qmk-keymap/blob/main/features/caps_word.h#L59-L68)).